### ISSUE TITLE
+ 2x speed On the new rail. Passage of the rail junction

### ISSUE THREAD

**Issue by Kellmyru** [TeamOpenIndustry/ImmersiveRailroading#1377]
Hello. I have a problem in my server.
Version mod: last Immersive Railroading 1.12.2 1.9.1
Configuration file use: The config was not affected
**Server:** Tried and installed on different cores. Mohist/Magma/Catserver (Catserver very bad work mod)
**Description of the problem:** 
When you drive and move to a new rail, i.e. you pass the junction, the speed doubles (or adds up, it is difficult to determine).
**The problem is not affected:**
Which player is riding on the train. Who installed the rails. It does not depend on the type of rails and their length, slope, block on which the rails stand. It does not depend on the amount of fuel, the number of cars in the train. Does not depend on the biome or terrain of the map.
Testing took place on different launchers. Including on a clean client without an optifine and with it.
There is also an observation that the further away the rail is from the player, the more likely another graphic bug will occur due to the fact that the textures of the rail, namely its embankment, will not be displayed until you put a block next to it or interact with the rail.
photo: 
![image](https://user-images.githubusercontent.com/112304022/236165712-8ae0a50d-9a14-4dd9-a2b3-50ba236bcc90.png)
![image](https://user-images.githubusercontent.com/112304022/236165902-0825649d-dd1e-4122-878d-8fefcfc96bd7.png)
![image](https://user-images.githubusercontent.com/112304022/236165996-15287cf2-4c00-4911-b701-3590e89e2c15.png)
If you don't stay in this chunk for a long time, such a small graphical bug will occur, I don't know if it can affect the situation with the speed of the train, but they didn't seem to notice the relationship.
We suspect that there is a part of the code that is responsible for assigning speed to the train when moving on a new section of rail, works 2 times and thus doubles the speed. The bug works as follows:
 1) Some distance, about 1-2 minutes drive, everything is fine. 
 2) When passing the rail junction, a bug occurs and the speed becomes doubly high.
 3) When moving such places, there may be small lags. 
 4) In the future, as soon as the bug occurred 1 time, it will occur at almost all junctions in the further path.
 5) If you start moving from another section, with a new train, the situation will repeat, but there are no guarantees that the bug will repeat at the same place. Ie, I want to note that the bug is not tied to a place on the map, but rather to the node or rail itself.
All cars and locomotives passed the test, including models not out of fashion. And the bug affected everyone the same way.
A resource pack is also installed on the server. But he does not add other wagons from the mod at the moment, but only changes the type of inventory of the player (as well as the client)
This problem its all players servers.
The problem persisted even when all modifications except Immersive Engineering and Immersive Railroading were removed

**System Information**  
OS: Windows 10
System Memory: 16 GB
Allocated Memory: 4GB and +
CPU: AMD Ryzen 7 4800H +radeon graph.
GPU: GTX 1660Ti
IR-Version: 1.12.2 1.9.1
Forge-Version: We tried different versions of forge. Mostly 14.23.5.2860
Java version client: java8 b352

**Video**
https://youtu.be/fU7KJLh2ZYs

**Modlist**  
 https://pastebin.com/cCRTxnBU
 
**latest.log**
 https://pastebin.com/rAC8zRVN

**Additional context**  
We tried to control both keys and handles with the mouse. There was a bug anyway. We even tried to change the control system in the config to a simplified one and vice versa, it didn't help.
It doesn't matter how fast the train is moving or what the player is doing at that moment. The bug happens after a while. Even on the straightest track in the flat world.

**Comment by Larky2k** [comment]
IR doesnt play nice on Hybrid servers, also throm those pics its just a visual bug when parts of rail bed isn't updated

**Comment by Kellmyru** [comment]
@Larky2k I agree with you about visual effects. In this appeal, I cited it as a possible cause of the main bug, although there seems to be no connection with this. I tried testing on a Regular forge. The problem appeared and went away again after a while. I think there is a vulnerability in the code that does not depend on the server core.

**Comment by Kellmyru** [comment]
After all, the speed is most likely added for a reason, which means there is duplication during the bug of certain actions that should not be. You can find this part of the code and make it more reliable. I presume

**Comment by Larky2k** [comment]
Its visual will have 0 effect on anything todo with IR

**Comment by Kellmyru** [comment]
I totally agree with you. I'm just ready to consider anything as the cause of the speed bug, because I can't cope on my own.

**Comment by Kellmyru** [comment]
Hello. I've tried testing on a clean forge 1.10.0 server, but the issue remains. 
In short, when building a long railway line (more than 1,000 blocks) and traveling along it, the server processes too many tiles, and when traveling through a rail junction, the current speed may double, and so on. I've tried different launchers, devices, and even server rental hosts, but I'm only using a clean forge server for testing. ( As you can see, I don't use plugins. )

**Comment by Kellmyru** [comment]
Yes, this happens in multiplayer . I will try your recomendet version. Thanks
